Source/engine.cpp

```diff
@@ -480,22 +480,24 @@ void CelBlitOutlineTo(CelOutputBuffer out, BYTE col, int sx, int sy, BYTE *pCelB
 				if (dst < out.end() && dst > out.begin()) {
 					if (dst >= out.end() - out.pitch()) {
 						while (width) {
-							if (*src++) {
+							if (!skipColorIndexZero || *src > 0) {
 								dst[-out.pitch()] = col;
 								dst[-1] = col;
 								dst[1] = col;
 							}
+							src++;

```

Comment:
Shouldn't we be incrementing this before the if, like the original code?

Reply:
It's a post-increment, so it was not before the if but after... this is why I don't like having them inside other operations, everyone gets confused :joy:

Reply:
Oh... indeed. I'm dumb.